spec/models/guide_search_indexer_spec.rb

```diff
@@ -1,47 +1,52 @@
 require 'rails_helper'
 
-RSpec.describe GuideSearchIndexer, "#index" do
-  it "indexes a document in rummager for the live edition" do
-    rummager_api = double(:rummager_api)
+RSpec.describe GuideSearchIndexer, '#index' do
+  before do
+    stub_any_rummager_post
+  end
+
+  it 'indexes a document in rummager for the live edition' do
     guide = create(:guide, :with_published_edition,
-                    title: "My guide",
+                    title: 'My guide',
                     body: "It's my published guide content",
-                    slug: "/service-manual/topic/some-slug"
+                    slug: '/service-manual/topic/some-slug'
                   )
     guide.editions << build(:edition, body: "I'm reconsidering this draft..")
 
-    expect(rummager_api).to receive(:add_document).with(
-      "service_manual_guide",
-      "/service-manual/topic/some-slug",
-      format:            "service_manual_guide",
-      description:       "Description",
+    described_class.new(guide).index
+
+    assert_rummager_posted_item(
+      _type:             'service_manual_guide',
+      _id:               '/service-manual/topic/some-slug',
+      format:            'service_manual_guide',
+      description:       'Description',
       indexable_content: "It's my published guide content",
-      title:             "My guide",
-      link:              "/service-manual/topic/some-slug",
-      manual:            "/service-manual",
-      organisations:     ["government-digital-service"],
+      title:             'My guide',
+      link:              '/service-manual/topic/some-slug',
+      manual:            '/service-manual',
+      organisations:     ['government-digital-service'],
     )
-
-    described_class.new(guide, rummager_api: rummager_api).index
   end
 
-  it "does not attempt to index a guide if it has no live editions" do
-    rummager_api = double(:rummager_api)
+  it 'does not attempt to index a guide if it has no live editions' do
     guide = create(:guide)
 
-    expect(rummager_api).to_not receive(:add_document)
+    described_class.new(guide).index
 
-    described_class.new(guide, rummager_api: rummager_api).index
+    assert_not_requested(
+      :post,
+      %r{#{GdsApi::TestHelpers::Rummager::RUMMAGER_ENDPOINT}/documents}
```

Comment:
It would be minorly more resilient to assert it doesn't call any Rummager URL. Not a biggie though. Whichever you prefer.